### PR TITLE
Do not log a warning if attempting a try-lock and it fails

### PR DESCRIPTION
**Pull request by mswintermeyer**
**Goals (and why)**:

Get rid of extraneous logging. It should not be too surprising if we fail to grab the lock in this scenario.

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java [hunk 1]
```diff
@@ -76,7 +76,16 @@ private static HeldLocksToken acquireLock(LockService lockService, LockRequest l
             if (response == null) {
                 RuntimeException ex = new LockAcquisitionException(
                         "Failed to lock using the provided lock request: " + lockRequest);
-                log.warn("Could not lock successfully", ex);
+                switch (lockRequest.getBlockingMode()) {
```
Comment (jkozlowski) [inline atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java]: nit: Put in a separate method called "handleCouldNotLock"

Comment (mswintermeyer) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java]: Now that I also handle failureCount (and logging when we surpass the limit), it's harder to extract. Thoughts?

Comment (jkozlowski) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java]: Well now it looks like you changed semantics? Previously you'd always retry, now you won't retry on DO_NOT_BLOCK (it feels correct, but I'm surprised there isn't a test that's failing now?)

Comment (mswintermeyer) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java]: Agreed that this changes semantics.. do you agree that these semantics are indeed what we want or were the old semantics somehow preferable?

Comment (jeremyk-91) [reply atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AdvisoryLockConditionSuppliers.java]: The new semantics are what I would expect and prefer overall. The concern is that a user using `DO_NOT_BLOCK` might end up failing more often, but I don't think that is actively a problem as previously these attempts would be clustered, and in the worst case the transaction will just need to try again.

**Comment by jeremyk-91** [comment]
Overriding status check for changelog, as it clearly exists.

**Comment by jkozlowski** [comment]
Can we followup with a test for this?

**Pull request merged**